libcfnet/tls_generic.c

```diff
@@ -724,7 +724,20 @@ int TLSRecv(SSL *ssl, char *buffer, int toget)
     assert(toget < CF_BUFSIZE);
     assert_SSLIsBlocking(ssl);
 
-    int received = SSL_read(ssl, buffer, toget);
+    int received,tries = 0,code=0;
+    do
+      {
+        received = SSL_read(ssl, buffer, toget);
+        if (received < 0)
+          {
+            code = TLSLogError(ssl, LOG_LEVEL_ERR, "SSL_read", received);
+            if (code == SSL_ERROR_WANT_READ)
```

Comment:
According to man:SSL_read(3), `SSL_Read()` can also return `SSL_ERROR_WANT_WRITE` so we need to check for that too.